dvc/logger.py

```diff
@@ -155,3 +156,65 @@ def debug(cls, msg, exc=None):
     @staticmethod
     def info(msg):
         Logger._with_progress(Logger.logger().info, msg)
+
+    @classmethod
+    def box(cls, msg, border_color=''):
+        lines = msg.split('\n')
+        max_line_length = max(len(cls._remove_colors(line)) for line in lines)
+
+        # Spaces between the borders and the content
+        padding_horizontal = 5
+        padding_vertical = 1
+
+        box_size_horizontal = (max_line_length + (padding_horizontal * 2))
+
+        chars = {
+            'top_left':     '┌',
+            'top_right':    '┐',
+            'bottom_right': '┘',
+            'bottom_left':  '└',
+            'vertical':     '│',
+            'horizontal':   '─',
+            'empty':        ' ',
+        }
+
+        top_line = "{left_corner}{line}{right_corner}\n".format(
+                        left_corner=chars['top_left'],
+                        line=chars['horizontal'] * box_size_horizontal,
+                        right_corner=chars['top_right']
+                    )
+
+        padding_lines = [
+            "{border}{padding}{content}{padding}{border}\n".format(
+                border=cls.colorize(chars['vertical'], border_color),
+                padding=chars['empty'] * padding_horizontal,
+                content=chars['empty'] * max_line_length,
+            ) * padding_vertical
+        ]
+
+        content_lines = [
+            "{border}{padding}{content}{padding}{border}\n".format(
+                border=cls.colorize(chars['vertical'], border_color),
+                padding=chars['empty'] * padding_horizontal,
+                content=line,
+            ) for line in lines
+        ]
+
+        bottom_line = "{left_corner}{line}{right_corner}\n".format(
+                          left_corner=chars['bottom_left'],
+                          line=chars['horizontal'] * box_size_horizontal,
+                          right_corner=chars['bottom_right']
+                      )
+
+        box = "{top}{padding}{content}{padding}{bottom}".format(
+            top=cls.colorize(top_line, border_color),
+            padding=''.join(padding_lines),
+            content=''.join(content_lines),
+            bottom=cls.colorize(bottom_line, border_color)
+        )
+
+        return box
+
+    @staticmethod
+    def _remove_colors(msg):
+        return re.compile(r'\x1b[^m]*m').sub('', msg)
```

Comment:
consider moving regex into a class constant, a separate method looks too much for such a simple function

Reply:
@shcheklein it is easier to read it this way though. I would leave it as is.

Reply:
It's minor. I meant something like:

REMOVE_COLORS = re.compile(r'\x1b[^m]*m')

just feels that function is too much in this case. And you need to be sure that Python is caching (it should be doing) compiled regexps, otherwise function gives you a penalty as well. Again, in this case it's all minor.